src/language-js/needs-parens.js

```diff
@@ -85,10 +86,11 @@ function needsParens(path, options) {
       !parent.optional
     ) {
       const statement = path.findAncestor(
-        (node) =>
-          node.type === "ExpressionStatement" ||
-          node.type === "ForStatement" ||
-          node.type === "ForInStatement"
+        createTypeCheckFunction([
+          "ExpressionStatement",
+          "ForStatement",
+          "ForInStatement",
+        ])
```

Comment:
I agree with using `createTypeCheckFunction` to create utility functions, but I think it's overkill to use it for an inline function that passes arguments like this one.

Reply:
I can add an option to only check top level ones. Just in case we still need inline ones in future.